### DISCUSSION TITLE
How to enter to and exit from GEM menus?

### DISCUSSION THREAD

**Discussion by MiqViq** [Spirik/GEM#95] [Q&A] [answered]
I am building a device that needs to show some basic info on the display, this info is not generated by GEM.
I would like enter GEM menu system only by using a specific "edit" button.
After that I would do some settings and save my changes, then exit from GEM menus to the default screen.
How to achieve this?

**Comment by Spirik** [comment, marked as answer]
Hi @MiqViq!

Recommended way to achieve this is through the use of [Context](https://github.com/Spirik/GEM#gemcontext). Normally context represents some functionality of the sketch outside of GEM (often it is actually the main functionality of the sketch). Usually user can load this context (i.e. run functionality) by pressing some button in menu. However in our case context should be loaded as soon as sketch starts. (And then it can be exited to GEM menu by press of the button - default `GEM_KEY_CANCEL` or by any other means implemented in context's loop routine as required by the author of the sketch).

Please, refer to [this modification](https://gist.github.com/Spirik/9d41973b13b6759ce0a2f8c5694f8771) of [Blink](https://github.com/Spirik/GEM/wiki/Example-02%3A-Blink) example (that loads context as soon as sketch starts) as a possible template of how to achieve desired behavior (modified lines denoted with `!!!` sign).

Hope that helps!
